### PR TITLE
preview infra support

### PR DESCRIPTION
**Pull request by fiji-flo**

### CODE REVIEW

File: buildspec-preview.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  pre_build:
+    commands:
+      - export COMMIT_SHA=$CODEBUILD_RESOLVED_SOURCE_VERSION
+      - export CODEBUILD_WEBHOOK_TRIGGER=${CODEBUILD_WEBHOOK_TRIGGER:-branch/preview}
+      - export PREVIEW=$(if [ "$CODEBUILD_WEBHOOK_TRIGGER" = "branch/preview" ]; then echo 1; fi)
+      - echo "running for ${COMMIT_SHA} in dev and test"
+      - aws ecr get-login --region us-west-2 --no-include-email | bash
+  build:
+    commands:
+      - if [ "$PREVIEW" = 1 ]; then myke docker --baseurl=/preview --name=dino-park-front-end-preview --rev=$COMMIT_SHA push-image --name=dino-park-front-end-preview --rev=$COMMIT_SHA; fi
```
Comment (The-smooth-operator) [inline buildspec-preview.yml]: --name and --rev are duplicated. Is this intended?

File: terraform/codebuild-preview/provider.tf [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#---
+# Provider Configuration
+#---
+
+provider "aws" {
+  region = "us-west-2"
+}
+
+terraform {
+  required_version = "~> 0.11"
```
Comment (The-smooth-operator) [inline terraform/codebuild-preview/provider.tf]: this conflicts with the last file in the PR which specifies >=0.12

File: buildspec-preview.yml (hunk 1 above)
Comment (fiji-flo) [reply buildspec-preview.yml]: Yes, they are arguments for the sub-commands once for `docker` once for `push-image`.

File: terraform/codebuild-preview/provider.tf (hunk 2 above)
Comment (fiji-flo) [reply terraform/codebuild-preview/provider.tf]: :+1:

File: buildspec-preview.yml (hunk 1 above)
Comment (The-smooth-operator) [reply buildspec-preview.yml]: :+1:

**Pull request merged**